packages/material-ui-based-components/src/components/TabsHorizontalComp.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext } from 'react';
+import { getRenderer, FormContext } from '@aemforms/af-react-renderer';
+import Tabs from '@mui/material/Tabs';
+import Tab from '@mui/material/Tab';
+// import Typography from '@mui/material/Typography';
+import Box from '@mui/material/Box';
+import { PROPS_PANEL } from '../utils/types';
+import { withRuleEnginePanel } from '../shared/withRuleEngine';
+
+interface TabPanelProps {
+    children?: any;
+    index: number;
+    value: number;
+}
+
+function TabPanel(props: TabPanelProps) {
+    const { children, value, index, ...other } = props;
+
+    return (
+        <div
+            role="tabpanel"
+            hidden={value !== index}
+            // id={`horizontal-tabpanel-${index}`}
+            id={children.id}
+            aria-labelledby={`horizontal-tab-${index}`}
+            {...other}
+        >
+            {value === index && (
+                <Box sx={{ p: 3 }}>
+                    {children}
+                </Box>
+            )}
+        </div>
+    );
+}
+
+function TabsHorizontalComp(props: PROPS_PANEL) {
+    const [value, setValue] = React.useState(0);
+    //@ts-ignore
+    const mappings = useContext(FormContext).mappings;
+    const items = props.items || [];
```

Comment:
Hidden items should not render here. Please filter props.item with the visible property.

Reply:
Applied the check on items as, items = props.visible ? props.items : []